### PR TITLE
update get-version to understand submodule tags

### PR DESCRIPTION
**Pull request by EvanBoyle**
Fixes https://github.com/pulumi/pulumi/issues/4227

### CODE REVIEW

File: scripts/get-version [hunk 1]
```diff
@@ -19,25 +20,42 @@ fi
...
+IFS='/'
+read -a tagsplit <<< "$TAG"
+
+if echo "${#tagsplit[@]}" > /dev/null 2>&1 -gt 1; 
+then
+    TAG=${tagsplit[${#tagsplit[@]}-1]}
+fi
+
+# if we found an exact match, return it at this point
+# after we have removed and submodule prefixes.
```
Comment (justinvp) [inline scripts/get-version]: Nit:

```suggestion
# If we found an exact match, return it at this point
# after we have removed any submodule prefixes.
```

File: scripts/get-version [hunk 2]
```diff
@@ -19,25 +20,42 @@ fi
...
     TAG=$(git describe --tags --abbrev=0 "${COMMITISH}")
 else
     TAG="v0.0.0"
 fi
 
+# Check to see if the latest tag is for a submodule and transform to semver tolerant
+# e.g: submodule/submodule/.../v.X.X.X
+# sdk/v1.1.1 -> v1.1.1
+IFS='/'
+read -a tagsplit <<< "$TAG"
```
Comment (justinvp) [inline scripts/get-version]: Nit: I guess it's not strictly necessary, but throughout this script we always use braces. Should we just use them here for consistency?

```suggestion
read -a tagsplit <<< "${TAG}"
```

File: scripts/get-version [hunk 3]
```diff
@@ -19,25 +20,42 @@ fi
...
+if echo "${#tagsplit[@]}" > /dev/null 2>&1 -gt 1; 
+then
+    TAG=${tagsplit[${#tagsplit[@]}-1]}
+fi
+
+# if we found an exact match, return it at this point
+# after we have removed and submodule prefixes.
+if [ $EXACT -eq 1 ]
+then
+    echo -n "$TAG"
```
Comment (justinvp) [inline scripts/get-version]: Nit: Similarly use braces here for consistency?

```suggestion
    echo -n "${TAG}"
```

**Comment by EvanBoyle** [comment]
Working on powershell updates. Need to spin up a windows WM to test them.

**Comment by EvanBoyle** [comment]
Tested the windows updates:

```
// before
PS C:\Users\Administrator\pulumi> .\scripts\get-version HEAD
pkg/v1.13.2-dev.1585432158+g6a945294
// after
PS C:\Users\Administrator\pulumi> .\scripts\get-version HEAD
v1.13.2-dev.1585522789+g4b8361d0
```

Also tried it with the script applied dirty on the tag commit:

```
PS C:\Users\Administrator\pulumi> .\scripts\get-version HEAD
v1.13.1+dirty
```

**Pull request merged**